### PR TITLE
add support to gcc8, add docker env test=develop

### PR DESCRIPTION
**Pull request by wopeizl**
add gcc8 support to paddle, including:
1. upgrade compiler to version 8+
2. upgrade the CUDA to version 10.1
3. fix the compile issue
4. add docker file for verification purpose

### CODE REVIEW

File: cmake/external/warpctc.cmake [hunk 1]
```diff
@@ -29,11 +29,7 @@ ELSE()
     SET(USE_OMP ON)
 ENDIF()
 
-IF(WIN32)
-    SET(WARPCTC_REPOSITORY "https://github.com/wopeizl/warp-ctc.git")
-ELSE()
-    SET(WARPCTC_REPOSITORY "https://github.com/dzhwinter/warp-ctc.git")
-ENDIF()
+SET(WARPCTC_REPOSITORY "https://github.com/wopeizl/warp-ctc.git")
```
Comment (JiabinYang) [inline cmake/external/warpctc.cmake:33]: Add a `TODO:` is better

File: tools/manylinux1/Dockerfile.GCC8 [hunk 2]
```diff
@@ -0,0 +1,178 @@
+# A image for building paddle binaries and install
```
Comment (JiabinYang) [inline tools/manylinux1/Dockerfile.GCC8:1]: add GCC8 to all environment

**Review by JiabinYang — COMMENTED** [review]
some comments

File: tools/manylinux1/Dockerfile.GCC8 (hunk 2 above)
Comment (wopeizl) [reply tools/manylinux1/Dockerfile.GCC8:1]: Gcc8 support cuda10.1 only for now.

File: cmake/external/warpctc.cmake (hunk 1 above)
Comment (wopeizl) [reply cmake/external/warpctc.cmake:33]: ok

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**